presto-docs/src/main/sphinx/spill/overview.rst

```diff
@@ -0,0 +1,85 @@
+.. contents::
+    :local:
+    :backlinks: none
+    :depth: 1
+
+Overview
+--------
+
+In the case of memory intensive operations, Presto allows offloading
+intermediate operator results to disk. The goal of this mechanism is to
+enable execution of queries that require amounts of memory exceeding per query
+or per node limits.
+
+The mechanism is similar to OS level page swapping. However, it is
+implemented on the application level to address specific needs of Presto.
+
+Properties related to spilling are described in :ref:`tuning-spilling`.
+
+
+Memory management and spill
+---------------------------
+
+By default, Presto kills queries if the memory requested by the query execution
+exceeds session properties ``query_max_memory`` or
+``query_max_memory_per_node``. This mechanism ensures fairness in allocation
+memory to queries and prevents deadlock caused by memory allocation.
+It is efficient when there is a lot of small queries in the cluster, but
+leads to killing large queries that don't stay within the limits.
+
+To overcome this inefficiency, the concept of revocable memory was introduced. A
+query can request memory that does not count toward the limits, but this memory
+can be revoked by the memory manager at any time. When memory is revoked, the
+query runner spills intermediate data from memory to disk and continues to
+process it later.
+
+In practice, when the cluster is idle, a memory intensive query may use all of
```

Comment:
It was not an intention.
Maybe it would be better to rephrase it as:
```
In practice, when the cluster is idle, and all memory is available, a memory intensive query may complete execution normally. On the other hand ....
```